docs/code-style-guide.md

```diff
@@ -62,6 +62,44 @@ The vast majority of the things noted in this document will apply to `C` code as
 
 * Functions that take no arguments must be declared as `foo(void)` **not** `foo()`, as the second option allows `foo` to take anything as it's arguments ([reference](https://softwareengineering.stackexchange.com/questions/286490/what-is-the-difference-between-function-and-functionvoid/286494))
 
+* Functions that return values via argument(s) must have all parameters labelled as `[in]`, `[in/out]`, or `[out]` in the javadoc, **in that order**.
+
+  ``` C
+  // Incorrect
+  /**
+   * Create a trajectory with given max speed
+   * 
+   * @param max_acceleration The maximum acceleration permitted. This will be
+   *                         updated to the maximum acceleration actually seen
+   *                         on the generated trajectory.
+   * @param created_trajectory A pointer that will be set to the created 
+   *                           created trajectory.
+   * @param max_speed The maximum speed on the trajectory permitted on the 
+   *                  trajectory.
+   * @return true if the trajectory was generated successfully, false otherwise
+   */
+  bool generateTrajectory(float* max_acceleration, 
+                          Trajectory* created_trajectory, 
+                          float max_speed);
+
+  // Correct
+  /**
+   * Create a trajectory with given max speed
+   * 
+   * @param max_speed [in] The maximum speed on the trajectory permitted on the 
+   *                       trajectory.
+   * @param max_acceleration [in/out] The maximum acceleration permitted. This 
+   *                                  will be updated to the maximum acceleration 
+   *                                  actually seen on the generated trajectory.
+   * @param created_trajectory A pointer that will be set to the created 
```

Comment:
`creted_trajectory` missing an `in` or `out` label

Reply:
@garethellis0 bumping this comment

Reply:
Changed to :
```
   * @param created_trajectory [out] A pointer that will be set to the created 
   *                                 created trajectory.
```